rpcs/blockService.go

```diff
@@ -285,6 +291,27 @@ func (bs *BlockService) handleCatchupReq(ctx context.Context, reqMsg network.Inc
 	return
 }
 
+func (bs *BlockService) redirectRequest(round uint64, response http.ResponseWriter, request *http.Request) (err error) {
+
+	peers := bs.net.GetPeers(network.PeersPhonebookArchivers)
+	if len(peers) == 0 {
+		return fmt.Errorf("redirecRequest: no arcihver peers found")
+	}
+	peer := peers[0]
+	httpPeer, validHTTPPeer := peer.(network.HTTPPeer)
+	if !validHTTPPeer {
+		return fmt.Errorf("redirecRequest: error getting an http peer")
+	}
+	parsedURL, err := network.ParseHostOrURL(httpPeer.GetAddress())
+	if err != nil {
+		return err
+	}
+	parsedURL.Path = bs.net.SubstituteGenesisID(path.Join(parsedURL.Path, "/v1/{genesisID}/block/"+strconv.FormatUint(round, 36)))
```

Comment:
implement a method here called `FormatBlockQuery` and use it both for this line as well as the one in the universalFetcher.